prefect_dbt/cli/commands.py

```diff
@@ -157,3 +164,193 @@ def trigger_dbt_cli_command_flow():
+        target_configs = SnowflakeTargetConfigs(connector=snowflake_connector)
+        dbt_cli_profile = DbtCliProfile(
+            name="jaffle_shop",
+            target="dev",
+            target_configs=target_configs,
+        )
+        profile = dbt_cli_profile.get_profile()
+        dbt_init = DbtCoreOperation(
+            commands=["dbt debug", "dbt list"],
+            dbt_cli_profile=dbt_cli_profile,
+            overwrite_profiles=True
+        )
+        dbt_init.run()
+        ```
+
+        Execute a longer-lasting dbt run as a context manager.
+        ```python
+        with DbtCoreOperation(commands=["dbt run"]) as dbt_run:
+            dbt_process = dbt_run.trigger()
+            # do other things
+            dbt_process.wait_for_completion()
+            dbt_output = dbt_process.fetch_result()
+        ```
+    """
+
+    _block_type_name = "dbt Core Operation"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/5zE9lxfzBHjw3tnEup4wWL/9a001902ed43a84c6c96d23b24622e19/dbt-bit_tm.png?h=250"  # noqa
+    _documentation_url = "https://prefecthq.github.io/prefect-dbt/cli/commands/#prefect_dbt.cli.commands.DbtCoreOperation"  # noqa
+
+    profiles_dir: Optional[Path] = Field(
+        default=None,
+        description=(
+            "The directory to search for the profiles.yml file. "
+            "Setting this appends the `--profiles-dir` option to the dbt commands "
+            "provided. If this is not set, will try using the DBT_PROFILES_DIR "
+            "environment variable, but if that's also not "
+            "set, will use the default directory `$HOME/.dbt/`."
+        ),
+    )
+    project_dir: Optional[Path] = Field(
+        default=None,
+        description=(
+            "The directory to search for the dbt_project.yml file. "
+            "Default is the current working directory and its parents."
+        ),
+    )
+    overwrite_profiles: bool = Field(
+        default=False,
+        description=(
+            "Whether the existing profiles.yml file under profiles_dir "
+            "should be overwritten with a new profile."
+        ),
+    )
+    dbt_cli_profile: Optional[DbtCliProfile] = Field(
+        default=None,
+        description=(
+            "Profiles class containing the profile written to profiles.yml. "
+            "Note! This is optional and will raise an error if profiles.yml already "
+            "exists under profile_dir and overwrite_profiles is set to False."
+        ),
+    )
+
+    @root_validator(pre=True)
+    def _check_installation(cls, values):
+        """
+        Check that dbt is installed.
+        """
+        if not which("dbt"):
+            raise ImportError(
+                "dbt-core needs to be installed to use this task; run "
+                '`pip install "prefect-dbt[cli]"'
+            )
+        return values
+
+    @root_validator
+    def _check_profiles(cls, values):
```

Comment:
I think this profile management should happen directly before the commands are invoked instead of when the object is instantiated. If changes are made between when the object is instantiated and when the commands are run, those changes wouldn't be respected.